File: app/src/main/java/com/iutcalendar/event/EventRecycleView.java
package com.iutcalendar.event;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.calendar.iutcalendar.R;
import com.iutcalendar.calendrier.EventCalendrier;
import com.iutcalendar.swiping.GestureEventManager;
import com.iutcalendar.swiping.TouchGestureListener;
import com.iutcalendar.task.PersonnalCalendrier;

import java.util.List;

public class EventRecycleView extends RecyclerView.Adapter<EventViewHolder> {

    List<EventCalendrier> list;
    EventViewHolder viewHolder;

    Context context;
    ClickListener clickListener;

    public EventRecycleView(List<EventCalendrier> list, Context context, ClickListener clickListener) {
        this.list = list;
        this.context = context;
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public EventViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        View eventView = inflater.inflate(R.layout.event_card, parent, false);
        viewHolder = new EventViewHolder(eventView);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull EventViewHolder holder, int position) {
        final int index = viewHolder.getAbsoluteAdapterPosition();
        EventCalendrier e = list.get(position);
        viewHolder.debut.setText(e.getDate().timeToString());
        viewHolder.fin.setText(e.getDate().addTime(e.getDuree()).timeToString());
        viewHolder.summary.setText(e.getSummary());
        viewHolder.salle.setText(e.getSalle().replace("\\,", "\n"));


        int numberTask = PersonnalCalendrier.getInstance(context).getCountTaskOf(e);
        if (numberTask == 0) {
            viewHolder.countTask.setVisibility(View.INVISIBLE);
        } else {
            viewHolder.countTask.setText(String.valueOf(numberTask));
        }
        viewHolder.view.setOnTouchListener(new TouchGestureListener(context, new GestureEventManager() {
            @Override
            protected void onClick() {
                clickListener.click(index);
                super.onClick();
            }

            @Override
            public void onSwipeRight() {
                super.onSwipeRight();
            }

            @Override
            public void onSwipeLeft() {
                super.onSwipeLeft();
            }

            @Override
            public void onSwipeDown() {
                super.onSwipeDown();
            }
        }));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }
}

File: app/src/main/java/com/iutcalendar/event/EventRecycleView.kt
package com.iutcalendar.event

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.iutcalendar.calendrier.EventCalendrier
import com.iutcalendar.swiping.GestureEventManager
import com.iutcalendar.swiping.TouchGestureListener
import com.iutcalendar.task.PersonalCalendrier
import com.univlyon1.tools.agenda.R

class EventRecycleView(var list: List<EventCalendrier?>?, var context: Context, var clickListener: (Int) -> Unit) : RecyclerView.Adapter<EventViewHolder>() {
    private var viewHolder: EventViewHolder? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EventViewHolder {
        val context = parent.context
        val inflater = LayoutInflater.from(context)
        val eventView = inflater.inflate(R.layout.event_card, parent, false)
        viewHolder = EventViewHolder(eventView)
        return viewHolder!!
    }

    override fun onBindViewHolder(holder: EventViewHolder, position: Int) {
        val index = viewHolder!!.absoluteAdapterPosition
        val e = list!![position]!!
        viewHolder!!.debut.text = e.date?.timeToString()
        viewHolder!!.fin.text = e.date!!.addTime(e.dure).timeToString()
        viewHolder!!.summary.text = e.nameEvent
        viewHolder!!.salle.text = e.salle.replace("\\,", "\n")
        val numberTask: Int = PersonalCalendrier.getInstance(context)!!.getCountTaskOf(e)
        if (numberTask == 0) {
            viewHolder!!.countTask.visibility = View.INVISIBLE
        } else {
            viewHolder!!.countTask.text = numberTask.toString()
        }
        viewHolder!!.view.setOnTouchListener(TouchGestureListener(context, object : GestureEventManager() {
            override fun onClick() {
                clickListener(index)
                super.onClick()
            }
        }))
    }

    override fun getItemCount(): Int {
        return list!!.size
    }
}